lib/Selenium/Remote/Driver.pm

```diff
@@ -866,13 +866,14 @@ sub new_session {
     my $args = {
         'desiredCapabilities' => {
             'browserName'        => $self->browser_name,
-            'platform'           => $self->platform,
             'javascriptEnabled'  => $self->javascript,
             'version'            => $self->version,
             'acceptSslCerts'     => $self->accept_ssl_certs,
             %$extra_capabilities,
         },
     };
+    $args->{'desiredCapabilities'}{'platform'} = $self->platform
+        if $self->browser_name ne 'firefox' || !$self->{is_wd3};
```

Comment:
what if we actually want to pass a platform other than 'ANY'? does that also cause the geckodriver to die?

Reply:
I tried imposing a few, same results.

Reply:
Dang.  Sounds like a proper bug in the driver then.